drv/hf-api/src/lib.rs

```diff
@@ -101,4 +101,92 @@ pub struct HfPersistentData {
     pub dev_select: HfDevSelect,
 }
 
+/// Represents persistent data that is both stored on the host flash and used to
+/// configure host boot.
+///
+/// We reserve sector 0 (i.e. the lowest 64 KiB) of both host flash ICs for
+/// Hubris data.  On both host flashes, we tile sector 0 with this 40-byte
+/// struct, placed at 128-byte intervals (in case it needs to grow in the
+/// future).
+///
+/// The current value of persistent data is the instance of `RawPersistentData`
+/// with a valid checksum and the highest `monotonic_counter` across both flash
+/// ICs.
+///
+/// When writing new data, we increment the monotonic counter and write to both
+/// ICs, one by one.  This ensures robustness in case of power loss.
+#[derive(Copy, Clone, Eq, PartialEq, AsBytes, FromBytes)]
+#[repr(C)]
+pub struct HfRawPersistentData {
+    /// Reserved field, because this is placed at address 0, which PSP firmware
+    /// may look at under certain circumstances.
+    amd_reserved_must_be_all_ones: u64,
+
+    /// Must always be `HF_PERSISTENT_DATA_MAGIC`.
+    oxide_magic: u32,
+
+    /// Must always be `HF_PERSISTENT_DATA_HEADER_VERSION` (for now)
+    header_version: u32,
+
+    /// Monotonically increasing counter
+    pub monotonic_counter: u64,
+
+    /// Either 0 or 1; directly translatable to [`HfDevSelect`]
+    pub dev_select: u32,
```

Comment:
I'm going to not touch this, because it's already used for Gimlet host-flash stuff (and we can't derive `FromBytes` on a `HfDevSelect`, because it could have invalid bit paterns)